drivers/pse/si3474/si3474.c

```diff
@@ -0,0 +1,383 @@
+#define DT_DRV_COMPAT skyworks_si3474
+
+#include <stdbool.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/pse.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+
+#include "si3474.h"
+#include "si3474_setup.h"
+#include "si3474_reg.h"
+#include "si3474_setup.h"
+
+LOG_MODULE_REGISTER(SI3474, CONFIG_PSE_LOG_LEVEL);
+
+int si3474_i2c_write_reg(const struct device *i2c, uint16_t dev_addr, uint8_t reg_ddr, uint8_t val)
+{
+	int res = i2c_reg_write_byte(i2c, dev_addr, reg_ddr, val);
+	if (res != 0) {
+		LOG_WRN("Si3474 device 0x%x writing 0x%x failed! [%i]", dev_addr, reg_ddr, res);
+		return res;
+	}
+	return res;
+}
+
+int si3474_i2c_read_reg(const struct device *i2c, uint16_t dev_addr, uint8_t reg_ddr, uint8_t *buff)
+{
+	int res = 0;
+	res = i2c_reg_read_byte(i2c, dev_addr, reg_ddr, buff);
+	if (res != 0) {
+		LOG_WRN("Si3474 device 0x%x reading 0x%x failed...! [%i]", dev_addr, reg_ddr, res);
+		return res;
+	}
+	return res;
+}
+
+static int si3474_get_events(const struct device *dev, uint8_t *events)
+{
+	const struct si3474_config *cfg = dev->config;
+	uint16_t dev_address = cfg->i2c.addr;
+	uint8_t pse_interrupt_buffer;
+	int res = 0;
+	int mask = 0x1;
+
+	res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, INTERRUPT_REGISTER,
+				  &pse_interrupt_buffer);
+	if (res != 0) {
+		LOG_WRN("Si3474 reading interrupt status failed");
```

Comment:
Do you still want to continue if you failed to read the interrupt status?

Reply:
For sure - must be corretced.